packages/vx-xychart/src/hooks/useScales.ts

```diff
@@ -0,0 +1,90 @@
+import { AxisScaleOutput, AxisScale } from '@vx/axis';
+import { ScaleConfig, NumberLike, createScale, ScaleInput } from '@vx/scale';
+import { extent as d3Extent } from 'd3-array';
+import { useMemo } from 'react';
+import DataRegistry from '../classes/DataRegistry';
+
+/** A hook for creating memoized x- and y-scales. */
+export default function useScales<
+  XScale extends AxisScale,
+  YScale extends AxisScale,
+  Datum = unknown
+>({
+  xScaleConfig,
+  yScaleConfig,
+  dataRegistry,
+  xRange,
+  yRange,
+}: {
+  xScaleConfig: ScaleConfig<AxisScaleOutput>;
+  yScaleConfig: ScaleConfig<AxisScaleOutput>;
+  dataRegistry: DataRegistry<XScale, YScale, Datum>;
+  xRange: [number, number];
+  yRange: [number, number];
+}) {
+  // pull out memoization keys that are less likely to change
+  const registryKeys = dataRegistry.keys();
+  const [xMin, xMax] = xRange;
+  const [yMin, yMax] = yRange;
+
+  const memoizedXScale = useMemo(() => {
+    const registryEntries = registryKeys.map(key => dataRegistry.get(key));
+
+    let xScale = createScale(xScaleConfig) as XScale;
```

Comment:
note: when passing `Scale` in as the generic, the only way I could this right was to coerce here. while we can infer `Scale` from `ScaleConfig`, we can't currently do the opposite.